.github/workflows/test.yaml

```diff
@@ -31,27 +33,56 @@ jobs:
         pip install nb-clean
         find book -name "*ipynb" | xargs nb-clean check
 
-
   build-and-test:
     runs-on: ubuntu-20.04
+    # This workflow accesses secrets and checks out a PR, so only run if labelled
+    # https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+    if: contains(github.event.pull_request.labels.*.name, 'preview')
 
     steps:
     - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Setup Cache
+      uses: actions/cache@v2
+      # NOTE: change key to "jupyterbook-N+1" to force rebuilding cache
+      with:
+        path: ./book/_build
+        key: jupyterbook-4
 
     - name: Pull Docker Image
       run: |
         docker pull $DOCKER_IMAGE
         docker images
 
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-west-2
+        role-to-assume: github-actions-role
+        role-duration-seconds: 1800
```

Comment:
Yes! it'd be nice to keep things DRY :) looks like it might be possible with new syntax made available last month? https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/ would probably require some experimentation to get right, but could open a separate issue to track

Reply:
gosh, every time i look at the docs i feel like there are tons of new options! Let's definitely punt on this, i'll open a new issue with some relevant links.

Reply:
see #17